Clean/Circomlib/Bitify2.lean

```diff
@@ -135,8 +149,25 @@ def circuit : FormalCircuit (F p) (fields 254) field where
     output.val = fromBits (input.map ZMod.val)
 
   soundness := by
-    simp only [circuit_norm, main, Bits2Num.main]
-    sorry
+    intro i0 env input_var input h_input assumptions h_holds
+    simp only [circuit_norm, main, Bits2Num.main] at h_holds ⊢
+    simp_all only [circuit_norm, AliasCheck.circuit]
+    simp only [id_eq] at h_holds
+    obtain ⟨ h_bits, h_eq ⟩ := h_holds
+    rw[← ZMod.val_natCast_of_lt h_bits]
+    rw[← mapFinRange_eq_map]
+    rw[← fieldFromBits_eq_mapFinRange_cast]
+    conv =>
+          rhs
+          congr
+          congr
+          congr
+          ext i
+          rw [← h_input]
+          simp
```

Comment:
When `simp` doesn't close a goal completely, we prefer `simp only [...]`. This makes the script robust against `simp` annotation changes.
```suggestion
          simp only [Fin.getElem_fin, Vector.getElem_map]
```